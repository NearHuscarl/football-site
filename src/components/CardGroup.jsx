import React from 'react';
import StandingCard from './StandingCard';

const CardGroup = () => (
    <div className='content-container'>
        <StandingCard />
    </div>
);

export default CardGroup;