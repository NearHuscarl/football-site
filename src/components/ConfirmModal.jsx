import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

const ConfirmModal = (props) => {
	const { isOpen, onRequestClose, prompt, onConfirm, onCancel } = props;
	return (
		<Modal
			isOpen={isOpen}
			onRequestClose={onRequestClose}
			contentLabel="Confirm Delete"
			closeTimeoutMS={200}
			className="modal"
			ariaHideApp={false}>
			<h3 className="modal__title">{prompt}</h3>
			<button
				type="button"
				className="button modal__button"
				onClick={onConfirm}>
				yes
			</button>
			<button
				type="button"
				className="button modal__button"
				onClick={onCancel}>
				no
			</button>
		</Modal>
	);
};

ConfirmModal.propTypes = {
	isOpen: PropTypes.bool.isRequired,
	onRequestClose: PropTypes.func,
	prompt: PropTypes.string.isRequired,
	onConfirm: PropTypes.func.isRequired,
	onCancel: PropTypes.func.isRequired,
};

ConfirmModal.defaultProps = {
	onRequestClose: () => {},
};

export default ConfirmModal;
