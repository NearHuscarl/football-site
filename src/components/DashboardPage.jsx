import React from 'react';

const ExpenseDashboardPage = () => <div>Dashboard page content</div>;

export default ExpenseDashboardPage;
