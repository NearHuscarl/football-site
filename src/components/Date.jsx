import React from 'react';
import { SingleDatePicker } from 'react-dates';

class Date extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            focused: null,
		};
    }

    onFocusChange = ({ focused }) => {
        this.setState(() => ({ focused }));
    }

    render() {
        return (
            <div className="date-wrapper">
                <SingleDatePicker
                    id=''
                    focused={this.state.focused}
                    onFocusChange={this.onFocusChange}
                    isOutsideRange={(day) => false}
                    numberOfMonths={1}
                    {...this.props}
                />
            </div>
        );
    }
}

export default Date;