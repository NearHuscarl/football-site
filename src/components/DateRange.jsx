import React from 'react';
import { DateRangePicker } from 'react-dates';

class DateRange extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
			focused: null,
		};
    }
    
	onFocusChange = (focused) => {
		this.setState(() => ({ focused }));
	};

    render() {
        return (
            <div className="date-wrapper">
                <DateRangePicker
                    startDateId=''
                    endDateId=''
                    focusedInput={this.state.focused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={(day) => false}
                    {...this.props}
                />
            </div>
        );
    }
}

export default DateRange;