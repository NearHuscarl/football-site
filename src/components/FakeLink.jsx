import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const fakeLinkStyle = {
	textDecoration: 'none',
	color: 'inherit',
};

const FakeLink = (props) => (
	<Link style={fakeLinkStyle} {...props}>{props.children}</Link>
);


FakeLink.propTypes = {
	children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
};

export default FakeLink;