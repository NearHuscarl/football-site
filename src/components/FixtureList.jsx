import React from 'react';
import { connect } from 'react-redux';
import isEmpty from 'lodash/isEmpty';
import FixtureListItem from './FixtureListItem';
import Loader from './Loader';
import { startSearchMatches } from '../actions/matchResults';
import { competitions } from '../settings';

// TODO: convert into presentation props (move connect upward)
class FixtureList extends React.Component {
    constructor(props) {
        super(props);
        this.props.startSearchMatches();
    }

    renderFixtureItem = (fixture) => {
        const { teams } = this.props;
        const competitionId = fixture.competition.id;
        const homeId = fixture.homeTeam.id;
        const awayId = fixture.awayTeam.id;
        if (isEmpty(teams[competitionId])) {
            return null;
        }
        const homeTeam = teams[competitionId][homeId];
        const awayTeam = teams[competitionId][awayId];

        return (
            <FixtureListItem
                key={fixture.id}
                fixture={fixture}
                homeTeam={homeTeam}
                awayTeam={awayTeam} />
        );
    }

    renderFixtureGroup = (fixtures) => {
        const competitionName = fixtures[0].competition.name;

        return (
            <div key={competitionName}>
                <div className='fixturelist-groupname'>
                    {competitionName}
                </div>
                {
                    fixtures.map((fixture) => this.renderFixtureItem(fixture))
                }
            </div>
        );
    }

    isDataReady = () => {
        const { teams } = this.props;

        for (const competitionId of Object.values(competitions)) {
            if (isEmpty(teams[competitionId])) {
                return false;
            }
        }
        return true;
    }

    renderFixtures = () => {
        const { fixtures } = this.props;
        let fixturesByCompetition = {};

        fixtures.forEach((fixture) => {
            const competitionId = fixture.competition.id;

            if (isEmpty(fixturesByCompetition[competitionId])) {
                fixturesByCompetition[competitionId] = [];
            }

            fixturesByCompetition[competitionId].push(fixture);
        });

        return Object.keys(fixturesByCompetition).map((competitionId) => {
            return this.renderFixtureGroup(fixturesByCompetition[competitionId]);
        });
    }

    render() {
        return (
            <div className='content-container'>
                {this.isDataReady() ?
                    this.renderFixtures()
                    :
                    <Loader height='40vh' />
                }
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    fixtures: state.matchResults.results,
    teams: state.teams,
})

const mapDispatchToProps = (dispatch) => ({
    startSearchMatches: () => {
        return dispatch(startSearchMatches());
    },
})

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(FixtureList);