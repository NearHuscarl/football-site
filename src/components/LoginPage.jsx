import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as authActions from '../actions/auth';

const LoginPage = (props) => {
	const {
		startLoginWithGoogle,
		startLoginWithFacebook,
		startLoginWithTwitter,
		startLoginWithGithub,
	} = props;

	return (
		<div className="box-layout">
			<div className="box-layout__box">
				<h1 className="box-layout__title">Boilerplate</h1>
				<p>Tag line for app.</p>
				<button
					type="button"
					className="box-layout__button button--red"
					onClick={startLoginWithGoogle}>
					<i className="fa fa-google fa-fw" />
					Login with Google
				</button>
				<button
					type="button"
					className="box-layout__button"
					onClick={startLoginWithFacebook}>
					<i className="fa fa-facebook fa-fw" />
					Login with Facebook
				</button>
				<button
					type="button"
					className="box-layout__button button--light-blue"
					onClick={startLoginWithTwitter}>
					<i className="fa fa-twitter fa-fw" />
					Login with Twitter
				</button>
				<button
					type="button"
					className="box-layout__button button--dark"
					onClick={startLoginWithGithub}>
					<i className="fa fa-github fa-fw" />
					Login with Github
				</button>
			</div>
		</div>
	);
};

LoginPage.propTypes = {
	startLoginWithGoogle: PropTypes.func.isRequired,
	startLoginWithFacebook: PropTypes.func.isRequired,
	startLoginWithTwitter: PropTypes.func.isRequired,
	startLoginWithGithub: PropTypes.func.isRequired,
};

export const MockLoginPage = LoginPage;

const mapDispatchToProps = (dispatch) => ({
	startLoginWithGoogle: () => dispatch(authActions.startLoginWithGoogle()),
	startLoginWithFacebook: () => dispatch(authActions.startLoginWithFacebook()),
	startLoginWithTwitter: () => dispatch(authActions.startLoginWithTwitter()),
	startLoginWithGithub: () => dispatch(authActions.startLoginWithGithub()),
});

export default connect(
	undefined,
	mapDispatchToProps,
)(LoginPage);
