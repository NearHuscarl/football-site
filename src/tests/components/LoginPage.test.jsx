import React from 'react';
import { shallow } from 'enzyme';
import { MockLoginPage } from '../../components/LoginPage';

let wrapper;
let startLoginWithGoogle;
let startLoginWithFacebook;
let startLoginWithTwitter;
let startLoginWithGithub;

beforeAll(() => {
	startLoginWithGoogle = jest.fn();
	startLoginWithFacebook = jest.fn();
	startLoginWithTwitter = jest.fn();
	startLoginWithGithub = jest.fn();
	wrapper = shallow(
		<MockLoginPage
			startLoginWithGoogle={startLoginWithGoogle}
			startLoginWithFacebook={startLoginWithFacebook}
			startLoginWithTwitter={startLoginWithTwitter}
			startLoginWithGithub={startLoginWithGithub}
		/>,
	);
});

test('should render LoginPage correctly', () => {
	expect(wrapper).toMatchSnapshot();
});

test('should call startLoginWithGoogle on button click', () => {
	wrapper
		.findWhere(
			(n) => n.text() === 'Login with Google' && n.type() === 'button',
		)
		.simulate('click');
	expect(startLoginWithGoogle).toHaveBeenCalled();
});

test('should call startLoginWithFacebook on button click', () => {
	wrapper
		.findWhere(
			(n) => n.text() === 'Login with Facebook' && n.type() === 'button',
		)
		.simulate('click');
	expect(startLoginWithFacebook).toHaveBeenCalled();
});

test('should call startLoginWithTwitter on button click', () => {
	wrapper
		.findWhere(
			(n) => n.text() === 'Login with Twitter' && n.type() === 'button',
		)
		.simulate('click');
	expect(startLoginWithTwitter).toHaveBeenCalled();
});

test('should call startLoginWithGithub on button click', () => {
	wrapper
		.findWhere(
			(n) => n.text() === 'Login with Github' && n.type() === 'button',
		)
		.simulate('click');
	expect(startLoginWithGithub).toHaveBeenCalled();
});
