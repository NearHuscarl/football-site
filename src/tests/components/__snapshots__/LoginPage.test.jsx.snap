// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
      Tag line for app.
    </p>
    <button
      className="box-layout__button button--red"
      onClick={[MockFunction]}
      type="button"
    >
      <i
        className="fa fa-google fa-fw"
      />
      Login with Google
    </button>
    <button
      className="box-layout__button"
      onClick={[MockFunction]}
      type="button"
    >
      <i
        className="fa fa-facebook fa-fw"
      />
      Login with Facebook
    </button>
    <button
      className="box-layout__button button--light-blue"
      onClick={[MockFunction]}
      type="button"
    >
      <i
        className="fa fa-twitter fa-fw"
      />
      Login with Twitter
    </button>
    <button
      className="box-layout__button button--dark"
      onClick={[MockFunction]}
      type="button"
    >
      <i
        className="fa fa-github fa-fw"
      />
      Login with Github
    </button>
  </div>
</div>
`;
